test/integration/integration_test.h

```diff
@@ -4,24 +4,25 @@
 
 #include "gtest/gtest.h"
 
-class IntegrationTest : public BaseIntegrationTest, public testing::Test {
+class IntegrationTest : public BaseIntegrationTest,
+                        public testing::TestWithParam<Network::Address::IpVersion> {
 public:
   /**
-   * Global initializer for all integration tests.
+   * Initializer for individual integration tests.
    */
-  static void SetUpTestCase() {
+  void SetUp() override {
```

Comment:
My impression was that removing state shared between tests was desired regardless of this effort. In that case, I was just using that fact in this PR. 

SetUpTestCase is only run once before the parameterized tests are instantiated so it doesn't have knowledge of the current IP address version parameter.